### DISCUSSION TITLE
svfbuveg

### DISCUSSION THREAD

**Discussion by langford77840** [UMEP-dev/UMEP#464] [General]
Hello,
I'd like to ask what are "svfbuveg, asvf, shmat, vegshmat, vbshvegshmat" variables and how we can calculate these variables.
It seems that these variables are new variables in the 2022 version compared to the previous version.

Thanks,

**Comment by biglimp** [comment]
svfbuveg is the total SkyViewFactor including hemispheric blocking from both buildings and vegetation.
@nilswallenberg can you help with the other four?

**Reply by langford77840** [reply]
Thank you so much for the quick response!
Are there any codes shared in this github that calculates svfbuveg?

**Comment by nilswallenberg** [comment]
Hi,

shmat, vegshmat and vbshvegshmat are matrices used when running SOLWEIG with an anisotropic sky, where the sky vault is divided into 153 patches. shmat indicates if a patch is sky or building, vegshmat if it is vegetation and vbshvegshmat if it is vegetation but with a building behind it. asvf is used to estimate if a patch made of a building surface is sunlit or in shade. 

You can read more here: [Lindberg & Grimmond, 2011](https://doi.org/10.1007/s00704-010-0382-8)

You can also read about the anisotropic sky and patches here: [Wallenberg et al., 2020](https://doi.org/10.1016/j.uclim.2020.100589)

Hopefully there will be another paper on the longwave component originating from the upper hemisphere available soon.

Cheers,
Nils

**Reply by langford77840** [reply]
Thank you so much for the quick response!
Are there any codes shared in this github that calculates shmat, vegshmat and vbshvegshmat ?
